weldx/geometry.py

```diff
@@ -2538,6 +2538,9 @@ def __post_init__(self, time):
                 add_dims=["n"],
             )
 
+        # make sure we have correct dimension order
+        self.coordinates = self.coordinates.transpose(..., "n", "c")
+
```

Comment:
this was causing the troubles with scan data transformation as an incorrectly transposed xarray **(p,c,n)** was passed to `SpatialData` @vhirtham

Reply:
Nice finding. I recently also passed a "misformed" xarray to ´SpatialData´ that gave me some headaches. The time coordinate was a datetime instead of a timedelta with an optional reference time. The error message was a bit cryptic and only occurred if time-dependent transformations were performed. The question is if we want to catch this too. While we might remember the format requirements, our new 2 users from the lab might get totally confused by such errors. Maybe we can introduce an ´xr_format_data_array´ function that orders dimensions, correct wrong time inputs and possibly other stuff.

Reply:
can you reproduce this bug?

the time values can be formatted here by using the `.weldx.time_ref_restore` xarray accessor, however all operations that rely on these time formats (e.g. interpolation) should also make sure to handle these internally so it would be good to know where this happens.

Reply:
Ideally, one can grasp the expected format from the function inputs

Reply:
~~~ python
import xarray as xr

from weldx import CoordinateSystemManager, SpatialData, Time, WXRotation

rot = WXRotation.from_euler("z", [0, 45, 90], True).as_matrix()

csm = CoordinateSystemManager("root")
csm.create_cs("workpiece", "root", coordinates=[0, 1, 2])
csm.create_cs(
    "TCP",
    "root",
    orientation=rot,
    coordinates=[[0, 0, 0], [1, 0, 0], [2, 0, 0]],
    time=Time(["2000", "2001", "2002"]),
)


da = xr.DataArray(
    [[[0, -1, 1], [0, 1, 1]], [[0, -1, 2], [0, 1, 2]], [[0, -1, -1], [0, 1, -1]]],
    dims=["time", "n", "c"],
    coords={"time": Time(["2000", "2001", "2002"]).as_pandas_index()},
)

sd = SpatialData(da)

csm.assign_data(sd, "data", "TCP", "workpiece")
~~~

This was the script I used to understand the error. I didn't create an issue cause I thought it was my own stupidity to violate the expected xarray format. However, it is probably better if those input violations are either handled or give an instant and clear response to the user as you said. The annoying thing here is: If you remover the instant transformation to the workpiece cs, the error happens later (for example during a `.plot` call or wherever the first transformation occurs) and getting the connection isn't that easy anymore.